lib/3scale/backend/stats/cleaner.rb

```diff
@@ -108,6 +98,25 @@ def delete!(redis_conns, log_deleted_keys: nil)
 
           private
 
+          def handle_redis_exception(exception, redis_conn)
+            # If it's a connection error, simply log so we can continue with
+            # other shards. If it's another kind of error, it could be caused by
+            # a bug, so better re-raise.
+
+            is_connection_err = REDIS_CONN_ERRORS.any? do |conn_error|
+              exception.is_a?(conn_error)
+            end
+
+            is_timeout = exception.is_a?(Redis::CommandError) &&
+              exception.message == 'ERR Connection timed out'.freeze
+
+            if is_connection_err || is_timeout
+              logger.error("Error while deleting stats of server #{redis_conn}: #{exception}")
+            else
+              raise exception
+            end
```

Comment:
What you want for this whole method is a `case` statement replicating the original `rescue` (it will also automatically apply `===` just like rescue).

Reply:
Changed. Although there's a little bit more duplication with the `case`, I think it's better because it avoids calling `===` directly.